internal/security/secret.go

```diff
@@ -28,20 +28,24 @@ import (
 
 // SecretProvider cache storage for the secrets
 type SecretProvider struct {
-	secretClient pkg.SecretClient
-	secrets      map[string]map[string]string
+	secretClient    pkg.SecretClient
+	secretsCache    map[string]map[string]string // secret's path, key, value
+	insecureSecrets *common.InsecureSecrets
+	loggingClient   logger.LoggingClient
 }
 
 // NewSecretProvider returns a new secret provider
 func NewSecretProvider() *SecretProvider {
-	return &SecretProvider{secrets: make(map[string]map[string]string)}
+	return &SecretProvider{secretsCache: make(map[string]map[string]string)}
 }
 
 // CreateClient creates a SecretClient to be used for obtaining secrets from a secrets store manager.
 func (s *SecretProvider) CreateClient(
 	loggingClient logger.LoggingClient,
 	configuration common.ConfigurationStruct) bool {
 
+	s.insecureSecrets = &configuration.Writable.InsecureSecrets
```

Comment:
I thought about this some more and this pointer is too deep into the structure... When updated from Consul, the whole Writable section is replaced and this pointer will be pointing to stale memory... :-( I think to correct way is to go back to a pointer to the complete configuration so when Writable is updated within it, we see the new values.